protoc-gen-twirp/generator.go

```diff
@@ -1334,8 +1334,15 @@ func (t *twirp) generateServerProtobufMethod(service *descriptor.ServiceDescript
 	t.P()
 	t.P(`  buf, err := `, t.pkgs["ioutil"], `.ReadAll(req.Body)`)
 	t.P(`  if err != nil {`)
-	t.P(`    s.writeError(ctx, resp, wrapInternal(err, "failed to read request body"))`)
-	t.P(`    return`)
+	t.P(`  		if ctxErr := context.Canceled; ctxErr == ctx.Err() {`)
+	t.P(`    		s.writeError(ctx, resp, twirp.NewError(twirp.Canceled, ctxErr.Error()))`)
+	t.P(`    		return`)
+	t.P(`  		} else if ctxErr := context.DeadlineExceeded; ctxErr == ctx.Err() {`)
+	t.P(`    		s.writeError(ctx, resp, twirp.NewError(twirp.DeadlineExceeded, ctxErr.Error()))`)
+	t.P(`    		return`)
+	t.P(`  		}`)
+	t.P(`    	s.writeError(ctx, resp, twirp.NewError(twirp.Malformed, "failed to read request body"))`)
```

Comment:
This case (returning the "malformed" code when the body read fails) is new since the first round. It's more of a behavior change but I think it's a good one, especially because the server returns that same status if the unmarshal step fails.

Reply:
Yes, Original reported came back and ask me for this and shared thread. So thought let me cover that too which I missed at first place.